samples/net/mqtt_publisher/src/main.c

```diff
@@ -452,6 +456,9 @@ static void publisher(void)
 
 		rc = process_mqtt_and_sleep(&client_ctx, APP_SLEEP_MSECS);
 		SUCCESS_OR_BREAK(rc);
+
+		LOG_INF("Waiting for keepalive timeout");
+		k_sleep(K_SECONDS(CONFIG_MQTT_KEEPALIVE + 1));
```

Comment:
I don't see a reason for this sleep. It will only stall the sample for 60 seconds (in the default configuration). 

The description in the commit is not accurate - the library will not send the ping in the background, as there's no such mechanism - you'd need to proactively call `mqtt_live` function afterwards. This sample just calls `mqtt_ping` on its own - which can be done at any time, no need to wait for keepalive timeout.

Reply:
My bad. I have found that the application keeps alive even without any sleep-code.
[00:00:00.110,000] <inf> net_config: Initializing network
[00:00:00.130,000] <inf> net_config: IPv4 address: 192.0.2.1
[00:00:03.180,000] <inf> net_mqtt_publisher_sample: attempting to connect: 
[00:00:03.190,000] <inf> net_mqtt_publisher_sample: MQTT client connected!
[00:00:03.190,000] <inf> net_mqtt_publisher_sample: try_to_connect: 0 <OK>
[00:00:03.190,000] <inf> net_mqtt_publisher_sample: mqtt_ping: 0 <OK>
**[00:00:03.200,000] <inf> net_mqtt_publisher_sample: PINGRESP packet**
[00:00:03.710,000] <inf> net_mqtt_publisher_sample: mqtt_publish: 0 <OK>

Now the sleep-code has been removed and commit message has been changed accordingly.